serialization/src/main/kotlin/maru/serialization/rlp/SealedBeaconBlockSerDe.kt

```diff
@@ -8,14 +8,17 @@
  */
 package maru.serialization.rlp
 
+import maru.compression.MaruCompressor
 import maru.core.SealedBeaconBlock
+import maru.serialization.compression.MaruSnappyFramedCompressor
 import org.hyperledger.besu.ethereum.rlp.RLPInput
 import org.hyperledger.besu.ethereum.rlp.RLPOutput
 
 class SealedBeaconBlockSerDe(
   private val beaconBlockSerializer: BeaconBlockSerDe,
   private val sealSerializer: SealSerDe,
-) : RLPSerDe<SealedBeaconBlock> {
+  private val compressor: MaruCompressor = MaruSnappyFramedCompressor(),
```

Comment:
Will this affect RLP going into the database? Generally better to let rocksdb do it's own compression.

Reply:
If we have rocksdb compression on, we may end up double compressing this and increasing the size for blocks in the database

Reply:
Checked we are using the default KVStoreConfiguration that has LZ4 compression enabled https://github.com/Consensys/teku/blob/master/storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/KvStoreConfiguration.java#L105 in the KvDatabaseFactory https://github.com/Consensys/maru/blob/main/storage/src/main/kotlin/maru/database/kv/KvDatabaseFactory.kt#L28

Reply:
Yes, now I'd reverted the changes on `SealedBeaconBlockSerDe` (just extends `RLPSerDe` instead of `MaruCompressorRLPSerDe` to disable compression), so the "db double compression" issue should be resolved. And introduced `SealedBeaconBlockCompressorSerDe` for compression when transmitting the gossip message